### ISSUE TITLE
Obscure error when executing empty distributed circuit

### ISSUE THREAD

**Issue by stavros11** [qiboteam/qibo#489]
**Describe the bug**
When executing a distributed circuit without any gates and attempting to access the final state, a hard to understand error appears.

**To Reproduce**
```Python
from qibo import gates, models
accelerators = {'/GPU:0': 1, '/GPU:1': 1}
c = models.Circuit(8, accelerators)
print(c().state())
```
gives
```
AttributeError: 'NoneType' object has no attribute 'list'
```

**Expected behavior**
Either print the correct final state which is |000...0> unless a different initial state was given, or raise a better error, eg. that distributed circuits without gates cannot be executed. Please let me know which would be preferrable.

**Comment by scarrazza** [comment]
I can reproduce this issue, looks like `self.qubits` is None, so most likely the `queue` attribute is not initialized properly before adding gates.

**Comment by stavros11** [comment]
> I can reproduce this issue, looks like `self.qubits` is None, so most likely the `queue` attribute is not initialized properly before adding gates.

Yes, I think the proper way to fix this is to make the circuit return the initial state instead of raising an error, because this is what happens for non-distributed circuits. I will open a PR for this.
